main.cpp

```diff
@@ -1,48 +1,574 @@
+{
+    return multiply( *dt.value );
+}
+
+FloatType& FloatType::divide(const IntType& it)
+{
+    return divide( *it.value );
+}
+
+FloatType& FloatType::divide(const FloatType& ft)
+{
+    return divide( *ft.value );
+}
+
+FloatType& FloatType::divide(const DoubleType& dt)
+{
+    return divide( *dt.value );
+}
+
+/* DoubleType member function definitions */
+
+DoubleType::DoubleType(double doublePrimitive) 
+{
+    value = new double(doublePrimitive);
+}
+
+DoubleType::~DoubleType() 
+{
+    delete value;
+}
+
+DoubleType& DoubleType::add(double d)
 {
-    std::cout << "good to go!" << std::endl;
+    *value += d;
+    return *this;
+}
+
+DoubleType& DoubleType::subtract(double d)
+{
+    *value -= d;
+    return *this;
+}
+
+DoubleType& DoubleType::multiply(double d)
+{
+    *value *= d;
+    return *this;
+}
+
+DoubleType& DoubleType::divide(double d)
+{
+    if (d == 0.0)
+    {
+        std::cerr << "warning: floating point division by zero!" << std::endl;
+        *value /= std::numeric_limits<double>::denorm_min();
+        return *this;
+    }
+    else
+    {
+        *value /= d;
+        return *this;
+    }
+}
+
+DoubleType& DoubleType::add(const IntType& it)
+{
+    return add( *it.value );
+}
+
+DoubleType& DoubleType::add(const FloatType& ft)
+{
+    return add( *ft.value );
+}
+
+DoubleType& DoubleType::add(const DoubleType& dt)
+{
+    return add( *dt.value );
+}
+
+DoubleType& DoubleType::subtract(const IntType& it)
+{
+    return subtract( *it.value );
+}
+
+DoubleType& DoubleType::subtract(const FloatType& ft)
+{
+    return subtract( *ft.value );
+}
+
+DoubleType& DoubleType::subtract(const DoubleType& dt)
+{
+    return subtract( *dt.value );
+}
+
+DoubleType& DoubleType::multiply(const IntType& it)
+{
+    return multiply( *it.value );
+}
+
+DoubleType& DoubleType::multiply(const FloatType& ft)
+{
+    return multiply( *ft.value );
+}
+
+DoubleType& DoubleType::multiply(const DoubleType& dt)
+{
+    return multiply( *dt.value );
+}
+
+DoubleType& DoubleType::divide(const IntType& it)
+{
+    return divide( *it.value );
+}
+
+DoubleType& DoubleType::divide(const FloatType& ft)
+{
+    return divide( *ft.value );
+}
+
+DoubleType& DoubleType::divide(const DoubleType& dt)
+{
+    return divide( *dt.value );
+}
+
+/* IntType member function definitions */
+
+IntType::IntType(int intPrimitive) 
+{
+    value = new int(intPrimitive);
+}
+
+IntType::~IntType() 
+{
+    delete value;
+}
+
+IntType& IntType::add(int i)
+{
+    *value += i;
+    return *this;
+}
+
+IntType& IntType::subtract(int i)
+{
+    *value -= i;
+    return *this;
+}
+
+IntType& IntType::multiply(int i)
+{
+    *value *= i;
+    return *this;
+}
+
+IntType& IntType::divide(int i)
+{
+    if (i <= 0.0)
+    {
+        std::cerr << "error: integer division by zero is an error and will crash the program!" << std::endl;
+        return *this;
+    } 
+    else 
+    {
+        *value /= i;
+        return *this;
+    }
+}
```

Comment:
D.R.Y. 
you're returning `*this` in both blocks.
- [ ] revise.

Reply:
Ok done revised divide member functions , tested committed.